odoo/addons/base/models/ir_actions_report.py

```diff
@@ -669,13 +676,19 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                             stream.close()
                             stream = new_stream
 
-                collected_streams[record.id] = {
+                collected_streams[res_id] = {
                     'stream': stream,
                     'attachment': attachment,
+                    'multiplicity': 1,
                 }
 
         # Call 'wkhtmltopdf' to generate the missing streams.
-        res_ids_wo_stream = [res_id for res_id, stream_data in collected_streams.items() if not stream_data['stream']]
+        res_ids_wo_stream = []
+        all_res_ids_wo_stream = []
+        for res_id, stream_data in collected_streams.items():
+            if not stream_data['stream']:
+                res_ids_wo_stream.append(res_id)
+                all_res_ids_wo_stream += [res_id] * stream_data.get('multiplicity', 1)
```

Comment:
=> `stream_data['multiplicity']`

Also because the usage of `multiplicity`, the order of res_ids is not respect, right ?

Reply:
@ryv-odoo Indeed. For now, let's just fix the "stock" case